### ISSUE TITLE
Drivers are missing

### ISSUE THREAD

**Issue by wangzhen0518** [files-community/Files#15373]
### Description

The all drivers of my computer are not showed in `Files`.

![Clip_2024-05-11_18-20-27](https://github.com/files-community/Files/assets/62546330/2a2af86d-e3eb-4bbe-8912-6605d578d09b)

I noticed issue #9774 is also about drivers' display. Here is the output of running cmd `powershell -c "Get-WmiObject Win32_LogicalDisk > '~\Desktop\disks.txt'; Get-WmiObject Win32_Volume >> '~\Desktop\disks.txt'; Get-WmiObject Win32_MountPoint >> '~\Desktop\disks.txt'"`

[disks.txt](https://github.com/files-community/Files/files/15282487/disks.txt)

### Steps To Reproduce

Just open `Files` and do nothing. When I click on "drivers"("驱动器"), nothing happens.

### Requirements

- e.g. Fix crash that occurs when...

### Files Version

3.4.0.0

### Windows Version

版本	Windows 11 专业版 版本	23H2 安装日期	‎2023/‎7/‎14 操作系统版本	22631.3527 体验	Windows Feature Experience Pack 1000.22700.1003.0

### Log File

[debug.log](https://github.com/files-community/Files/files/15282491/debug.log)

**Comment by Josh65-2201** [comment]
Thanks for the report, I've added this to the project board.

**Comment by yaira2** [comment]
Is this still an issue in the preview version? https://files.community/download/preview

**Comment by jdoescyber** [comment]
> Is this still an issue in the preview version? https://files.community/download/preview

Hi there. Not the OP, but I came to the Issues because I'm having the same problem.

Same issue in the current release as described, local drives don't show up at all. 

In the Preview release, my `C:` drive will show up after about 45 seconds, but none of the other local drives will.

```powershell
me@myredactedpc in ~
➜  Get-WmiObject Win32_LogicalDisk

DeviceID     : C:
DriveType    : 3
ProviderName :
FreeSpace    : 748087156736
Size         : 1023378714624
VolumeName   :

DeviceID     : D:
DriveType    : 3
ProviderName :
FreeSpace    : 182088155136
Size         : 1000186310656
VolumeName   : GAMES

DeviceID     : E:
DriveType    : 3
ProviderName :
FreeSpace    : 1399820283904
Size         : 2000381014016
VolumeName   : Data
```

Thanks for looking into this!

**Comment by 0x5bfa** [comment]
You OS drive letter is C?

---

For dev note:
We use System.IO.DriveInfo.GetDrives(); to enumerate drives.
I think we might as well use Win32API FindFirstVolume and FindNextVolume to see if this issue fixed or not.

**Comment by wangzhen0518** [comment]
> Is this still an issue in the preview version? https://files.community/download/preview

No, my drivers still cannot be shown in the preview version.

![Clip_2024-05-23_04-40-52](https://github.com/files-community/Files/assets/62546330/44a5a58d-78f1-4eb4-9b9c-65a9c4410c2d)


![Clip_2024-05-23_04-40-42](https://github.com/files-community/Files/assets/62546330/6a9816b4-f1c1-43ec-9494-991a07ad0a6a)

**Comment by 0x5bfa** [comment]
No no no my important question there

**Comment by wangzhen0518** [comment]
> No no no my important question there
> 
> Let us know if your OS drive letter is C?
> 
> Let us know the output of ‘mountvol /l’ in command prompt.


Yes, I installed win11 in driver C. The output of `mountvol /l` is as follows.

```shell
创建、删除或列出卷装入点。

MOUNTVOL [drive:]path VolumeName
MOUNTVOL [drive:]path /D
MOUNTVOL [drive:]path /L
MOUNTVOL [drive:]path /P
MOUNTVOL /R
MOUNTVOL /N
MOUNTVOL /E
MOUNTVOL drive: /S

    path        指定装入点将驻留的现有 NTFS 目录。
    VolumeName  指定装入点的目标的卷名称。
    /D          从指定的目录中删除卷装入点。
    /L          列出指定目录的已装入的卷名称。
    /P          从指定目录删除卷装入点，卸下此卷并使此卷无法装入。你可以创建
                一个卷来再次使此卷可以装入。
    /R          删除不在系统中的、卷的装入点目录和注册表设置。
    /N          禁用新卷的自动装入。
    /E          再次启用新卷的自动装入。
    /S          将 EFI 系统分区装载到提供的驱动器。

当前装入点的 VolumeName 可能值为:

    \\?\Volume{c06e52c4-2036-4009-9231-3760e6fa2674}\
        C:\

    \\?\Volume{e4f608b0-df6d-4d60-ad1e-b61596c2b561}\
        D:\

    \\?\Volume{56a37df9-a3ef-4c60-815e-61f0f11c3612}\
        *** 无装入点 ***

    \\?\Volume{69ea76ee-0453-416c-bb2e-2aedd693e04e}\
        Z:\

    \\?\Volume{287f7a08-3043-4520-9e0f-3293f7973530}\
        E:\

    \\?\Volume{112d35ae-3854-496f-ae36-76c854c90804}\
        *** 无装入点 ***
```

**Comment by 0x5bfa** [comment]
Great so using win32api should fix this issue.

**Comment by Josh65-2201** [comment]
@wangzhen0518 Is this still an issue on v4.0.11 https://files.community/download

**Comment by wangzhen0518** [comment]
No, Files works perfectly on my machine. This bug doesn't occur anymore. 👍

**Comment by Josh65-2201** [comment]
Thanks for checking
